gossip/src/cluster_info.rs

```diff
@@ -1571,6 +1564,8 @@ impl ClusterInfo {
             .map(|(_, filters)| filters.len())
             .sum::<usize>() as u64;
         self.stats.new_pull_requests_count.add_relaxed(num_requests);
+        // TODO: Use new ContactInfo once the cluster has upgraded to:
+        // https://github.com/anza-xyz/agave/pull/803
         let self_info = LegacyContactInfo::try_from(&self.my_contact_info())
             .map(CrdsData::LegacyContactInfo)
             .expect("Operator must spin up node with valid contact-info");
```

Comment:
Right, but that was only done in https://github.com/anza-xyz/agave/pull/803 which is not backported to v1.17 yet. So mainnet node still expect legacy contact-info in the pull request and otherwise drop the pull request.
To keep master branch compatible with mainnet, we cannot yet switch to the new contact-info here until mainnet has upgraded to #803 .

Reply:
is this the same reason we keep adding `LegacyContactInfo` to PullResponses with higher probability: https://github.com/behzadnouri/solana/blob/d6d95f92fbd311efabf51514062427d4dcb479fc/gossip/src/cluster_info.rs#L2124?